public/app/core/utils/UserProvider.tsx

```diff
@@ -78,6 +92,28 @@ export class UserProvider extends PureComponent<Props, State> {
     this.setState({ teams, loadingStates: { ...this.state.loadingStates, loadTeams: false } });
   };
 
+  loadOrgs = async () => {
+    this.setState({
+      loadingStates: { ...this.state.loadingStates, loadOrgs: true },
+    });
+    const orgs = await getBackendSrv().get('/api/user/orgs');
```

Comment:
Maybe try to move this async operation out of the Component into a Thunk, but that is just my personal opinion don't really know what our prefered way is. We're using a render prop pattern here to propagate values that are fetched in the backend. This ReactProfileWrapper could instead be connected to redux state where data is already in place.

Reply:
@hugohaggmark this is a conscious decision to use a provider like that, as the state in the settings pages is transient

Reply:
I had to try :)

Reply:
yea, agree that redux & thunk actions seem like a good fit for some of this. Just the transient form states that are easier to handle in the component.

Let's discuss this UserProvider pattern vs redux on a frontend weekly, good to evaluate and discuss pros / cons.

Reply:
Never awake for this :( ^